hubblestack/extmods/modules/nebula_osquery.py

```diff
@@ -335,32 +335,87 @@ def mask_passwords_inplace(object_to_be_masked):
     Note that this method alters "object_to_be_masked".
     '''
     try:
-        mask_file  = __salt__['cp.cache_file']('salt://hubblestack_nebula_v2/mask.yaml')
-        if not mask_file:
-          return
-        mask = yaml.safe_load(open(mask_file))
+
+        mask = {}
+        topfile = 'salt://hubblestack_nebula_v2/top.mask'
+        mask_files = get_top_data(topfile)
+        mask_files = ['salt://hubblestack_nebula_v2/' + mask_file.replace('.', '/') + '.yaml'
+                   for mask_file in mask_files]
+        if mask_files is None:
+            mask_files = 'salt://hubblestack_nebula_v2/mask.yaml'
```

Comment:
@MoodyMudit 
I have made some chages so that the top.mask can be passed as an argument. 
But still, the code contains some hardcodings. 
These hardcodings are present in the nebula code from a long time. I don't know the reason behind this.
Maybe @basepi  has some idea. 
But in a nutshell, if we ever move to hubblestack_nebula_v3, we must anyways make some other changes in the the nebula file. Hopefully, the masking code will also be changed accordingly at that time.